rapids-cmake/cpm/detail/load_preset_versions.cmake

```diff
@@ -34,15 +34,32 @@ Establish the `CPM` preset package information for the project.
 function(rapids_cpm_load_preset_versions)
   list(APPEND CMAKE_MESSAGE_CONTEXT "rapids.cpm.load_preset_versions")
 
+  set(_rapids_options)
+  set(_rapids_one_value PRESET_FILE)
+  set(_rapids_multi_value)
+  cmake_parse_arguments(_RAPIDS "${_rapids_options}" "${_rapids_one_value}"
+                        "${_rapids_multi_value}" ${ARGN})
+
+  set(_rapids_preset_version_file "${CMAKE_CURRENT_FUNCTION_LIST_DIR}/../versions.json")
+  if(_RAPIDS_PRESET_FILE)
+    set(_rapids_preset_version_file "${_RAPIDS_PRESET_FILE}")
+  endif()
+
+  if(NOT EXISTS "${_rapids_preset_version_file}")
+    message(FATAL_ERROR "rapids_cpm can't load '${filepath}' to find package version information, verify it exists"
+    )
+  endif()
+
   # Check if we have been loaded before, if so early terminate
-  get_property(already_loaded GLOBAL PROPERTY rapids_cpm_load_presets SET)
+  get_property(already_loaded GLOBAL PROPERTY rapids_cpm_load_presets_${_rapids_preset_version_file}
+               SET)
   if(already_loaded)
     return()
   endif()
-  set_property(GLOBAL PROPERTY rapids_cpm_load_presets "ON")
+  set_property(GLOBAL PROPERTY rapids_cpm_load_presets_${_rapids_preset_version_file} "ON")
```

Comment:
One thing to consider here is that the same file might be accessed by different relative paths, and thus will set two different properties. If this is merely a cache for optimization then it might not matter.